### PR TITLE
Fixed issue #16: Cannot display 2 reCaptcha on one page

### PR DESCRIPTION
**Pull request by magefan**

### CODE REVIEW

File: Block/Frontend/ReCaptcha.php [hunk 1]
```diff
@@ -90,7 +100,39 @@ public function getPublicKey()
...
     }
+
+    /**
+     * Get reCaptcha container unic ID
+     * @return string
+     */
+    public function getReCaptchaContainerId()
+    {
+        if (null === $this->reCaptchaContainerId) {
+            $this->reCaptchaContainerId = 'msp-recaptcha-container-' . md5($this->getNameInLayout());
```
Comment (tuyennn) [inline Block/Frontend/ReCaptcha.php]: @magefan The problem is caused by same scope on multiple area
So was you **missing** $this->getNameInLayout() on
`md5($this->getNameInLayout())` ?

And if in different layouts we have save name, would md5(a same value) could cover up the situation?

I think you should get each one different each others by defined different zone from layout, then md5 this from your **getNameInLayout** function
`<item name="zone" xsi:type="string">login-1</item>` 
`<item name="zone" xsi:type="string">login-2</item>` 
`<item name="zone" xsi:type="string">login-3</item>`

Comment (magefan) [reply Block/Frontend/ReCaptcha.php]: @tuyennn , we think that md5 from getNameInLayout should be enough as each block in Magento 2 has a unique name in the layout, so we don't need to use zone in addition.

**Comment by phoenix128** [comment]
Hello @magefan ,
sorry for the long delay in my answer.

Your PR looks good, but I am wondering how correct is to change the component name and scope at runtime.

I think a more proper solution would be to refactor `msp_recaptcha.phtml` and `MSP\ReCaptcha\Block\Frontend\ReCaptcha` in a deeper way allowing them to receive the component scope via layout XML and using `data-mage-init` instead of using `text/x-magento-init`.

Something like:
```
...
       <referenceContainer name="form.additional.info">
            <block class="MSP\ReCaptcha\Block\Frontend\ReCaptcha" name="msp-recaptcha" after="-"
                   template="MSP_ReCaptcha::msp_recaptcha.phtml">

                <arguments>
                    <argument name="scope" xsi:type="string">my-msp-recaptcha-scope</argument>
                    <argument name="component" xsi:type="array">
                        ...
                    </argument>
                    ...
                </arguments>
            </block>
        </referenceContainer>
...
```

```
...
<div class="field-recaptcha" data-mage-init='<?= ... ?>'>
    <!-- ko template: getTemplate() --><!-- /ko -->
</div>
...
```

**NOTE:**
Since this will break backward compatibility with layout, I think the proper way is to deprecate `msp_recaptcha.phtml` and `MSP\ReCaptcha\Block\Frontend\ReCaptcha` in favour of a new template and a ViewModel.

I think this would be a more clean solution.
Do you think you can rework your PR?

**Comment by magefan** [comment]
@phoenix128, yes we can rework PR, but we are concerning about ViewModel as a lot of clients still have Magento 2.1.x that does not support view models, or you are ready to stop supporting M2.1.x for this extension?

**Comment by phoenix128** [comment]
This was replaced by https://github.com/magento/magespecialist_ReCaptcha/pull/27 .

**Pull request closed without merging**